Formula/rizin.rb

```diff
@@ -0,0 +1,40 @@
+class Rizin < Formula
+  desc "UNIX-like reverse engineering framework and command-line toolset"
+  homepage "https://rizin.re"
+  url "https://github.com/rizinorg/rizin/releases/download/v0.3.1/rizin-src-v0.3.1.tar.xz"
+  sha256 "9ae4b7f9c47be63665fd0e59375b2b10f83c37156abb044ca4d61c1f0fc88f7e"
+  license "LGPL-3.0-only"
+  head "https://github.com/rizinorg/rizin.git", branch: "dev"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libuv"
+  depends_on "libzip"
+  depends_on "lz4"
+  depends_on "tree-sitter"
+
+  uses_from_macos "zlib"
+
+  def install
+    mkdir "build" do
+      args = [
```

Comment:
```suggestion
      args = [
        "-Dpackager=#{tap.user}",
        "-Dpackager_version=#{pkgversion}",
```

Why not also use Homebrew `libmagic` (`use_sys_magic`), `xxhash`, and `openssl`?

Reply:
Thanks 🙏 Seems to me that all extra folder will be related to either `--prefix` or `--libdir` (see [here](https://github.com/rizinorg/rizin/blob/906ece4211da45f0cd1259a0b258854f888dbbaa/meson.build#L186-L196)) so it should be ok if I understood correctly.
For `libmagic`, `rizin` provides an implementation that is either a [wrapper](https://github.com/rizinorg/rizin/blob/dev/librz/magic/magic.c#L20) (if `use_sys_magic` is enabled) or a custom implementation. Same goes for `capstone`, as there is a [fork](https://github.com/rizinorg/capstone) I thought it was best to use the bundled version.
For `xxhash`, it cannot be found by `pkg-config` during build, likely due to a mismatch in looking for `xxhash` and the file provided by brew called `libxxhash.pc`. I think it could be patched if this is accepted / preferred ?
For `openssl`, I just forgot 😅 

Thanks a lots for all reviews ! It's really helping as it is my first formula.

Reply:
Most other packagers (Arch, Fedora, Nix) use their own `libmagic` and `capstone`. I think we'll want to follow them here.

As for `xxhash`, this sounds like an issue for the Rizin project, since `libxxhash.pc` is the default that's generated by `xxhash`'s `Makefile`. As a temporary fix, we can do something like
```ruby
(buildpath/"pkgconfig").install_symlink Formula["xxhash"].opt_lib/"pkgconfig/libxxhash.pc" => "xxhash.pc"
ENV.prepend_path "PKG_CONFIG_PATH", buildpath/"pkgconfig"
```
at the start of the `install` method. We'll want to open an issue upstream about this too, though, to make sure we're not carrying this workaround forever.

As for the extra folders:

These directories are usually chosen under the assumption that packages are all installed into a common prefix (e.g. `/usr` or `/usr/local`) rather than what Homebrew does (installs each package into its own prefix and symlinks them into a common location). That makes some default install locations inappropriate for Homebrew. See, for example, [`gstreamer`](https://github.com/Homebrew/homebrew-core/blob/88510ce9b094a0cb4e8722a5f8744150e4733ae6/Formula/gstreamer.rb#L47-L48), [`vapoursynth`](https://github.com/Homebrew/homebrew-core/blob/88510ce9b094a0cb4e8722a5f8744150e4733ae6/Formula/vapoursynth.rb#L46), or even [`ruby`](https://github.com/Homebrew/homebrew-core/blob/88510ce9b094a0cb4e8722a5f8744150e4733ae6/Formula/ruby.rb#L66-L67).

I think the test here is whether we're okay with files installed into these directories to be deleted when someone does `brew upgrade rizin`. My guess is that things like [plugins or extras](https://github.com/rizinorg/rizin/blob/906ece4211da45f0cd1259a0b258854f888dbbaa/meson.build#L194-L195) shouldn't be deleted (those will probably be user-installed), and so those should reference `HOMEBREW_PREFIX` rather than the formula's `prefix`.

Does that make sense?

> Thanks a lots for all reviews ! It's really helping as it is my first formula.

No problem; we're always happy to help. Thanks for taking the time to contribute.

Reply:
By the way: if you're not sure about the extra folders issue, we can leave it be for now and fix it later if it does cause issues.

Reply:
> These directories are usually chosen under the assumption that packages are all installed into a common prefix (e.g. /usr or /usr/local) rather than what Homebrew does (installs each package into its own prefix and symlinks them into a common location). That makes some default install locations inappropriate for Homebrew. See, for example, gstreamer, vapoursynth, or even ruby.

Ok I now got your point on this, I think, indeed, it makes sense for `plugins`, `extras` and probably `bindings`. I added it similar to `ruby`.

On `libmagic`, I cannot make it build and I have trouble figuring out why `meson` is not finding the library. It seems related to using only `find_library` and not `dependency`.

Reply:
> On `libmagic`, I cannot make it build and I have trouble figuring out why `meson` is not finding the library. It seems related to using only `find_library` and not `dependency`.

Hmmm, that's odd. What's the error you get from `meson` about not being able to find `libmagic`? Maybe we can test it out on CI to see if it does better. Or, at least, I can see the error to work out if something can be done.

I'm also wondering if `find_library` is looking for a `magic.pc` when `libmagic` ships `libmagic.pc`. This would mean we can just use a workaround similar to what we're doing for `xxhash`.

No Homebrew `libmagic` isn't a huge dealbreaker though; we can probably punt on this with an upstream issue if we can't get it to work.

Reply:
I'm getting the following error during configuration with `meson`
> ../meson.build:232:0: ERROR: C shared library 'magic' not found

The tricks used for `xxhash` is not working as `find_library` is not using `pkgconfig`. Search logic is located [here](https://github.com/mesonbuild/meson/blob/f4b91c4306b68c200a54a53ba872903eed2fcd18/mesonbuild/compilers/mixins/clike.py#L1056) in `meson`. Procedure seems to have 2 steps:
* Try to compile and link some dummy code with `-llib`, if this fails
* Run the compiler with `--print-search-dirs` [option](https://github.com/mesonbuild/meson/blob/f4b91c4306b68c200a54a53ba872903eed2fcd18/mesonbuild/compilers/mixins/gnu.py#L276) and manually search into this path. Which on my system boils down to:
> programs: =/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
libraries: =/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0

Hence library not found 😕 I pushed a test commit for CI.